DFC.EventGridSubscriptions.ApiFunction/Function/DeadLetterHttpTrigger.cs

```diff
@@ -0,0 +1,120 @@
+        {
+            this.options = options;
+            this.subscriptionService = subscriptionService;
+        }
+
+        [FunctionName("ProcessDeadLetter")]
+        public async Task<HttpResponseMessage> Run([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "DeadLetter/api/updates")] HttpRequestMessage req, ILogger log)
+        {
+            Initialise(req);
+
+            log.LogInformation($"C# HTTP trigger function begun");
+            string response = string.Empty;
+
+            string requestContent = await req.Content.ReadAsStringAsync().ConfigureAwait(false);
+            log.LogInformation($"Received events: {requestContent}");
+
+            EventGridSubscriber eventGridSubscriber = new EventGridSubscriber();
+
+            EventGridEvent[] eventGridEvents = eventGridSubscriber.DeserializeEventGridEvents(requestContent);
+
+            foreach (EventGridEvent eventGridEvent in eventGridEvents)
+            {
+                if (eventGridEvent.Data.GetType() == typeof(SubscriptionValidationEventData))
+                {
+                    var eventData = eventGridEvent.Data as SubscriptionValidationEventData;
+
+                    if (eventData == null)
+                    {
+                        throw new InvalidDataException($"{nameof(SubscriptionValidationEventData)} in EventGridEvent {eventGridEvent.Id} is null");
+                    }
+
+                    log.LogInformation($"Got SubscriptionValidation event data, validation code: {eventData.ValidationCode}, topic: {eventGridEvent.Topic}");
+
+                    var responseData = new SubscriptionValidationResponse()
+                    {
+                        ValidationResponse = eventData.ValidationCode,
+                    };
+
+                    return new HttpResponseMessage(HttpStatusCode.OK)
+                    {
+                        Content = new StringContent(JsonConvert.SerializeObject(responseData), Encoding.UTF8, "application/json"),
+                    };
+                }
+                else if (eventGridEvent.Data.GetType() == typeof(StorageBlobCreatedEventData))
+                {
+                    if (options.CurrentValue.DeadLetterBlobContainerName == null)
+                    {
+                        throw new ArgumentException(nameof(options.CurrentValue.DeadLetterBlobContainerName));
+                    }
+
+                    var eventData = eventGridEvent.Data as StorageBlobCreatedEventData;
+
+                    if (eventData == null)
+                    {
+                        throw new InvalidDataException($"{nameof(StorageBlobCreatedEventData)} in EventGridEvent {eventGridEvent.Id} is null");
+                    }
+
+                    if (eventData.Url != null && eventData.Url!.ToUpperInvariant().Contains(options.CurrentValue.DeadLetterBlobContainerName, StringComparison.CurrentCultureIgnoreCase))
```

Comment:
Is the `ToUpperInvariant` required, as it passes the `StringComparison` to the `Contains`?

We might want to use `StringComparison.OrdinalIgnoreCase` to check the Url, as Url's should be culture agnostic. I find this [page](https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings) useful.

Also, as a general pattern, it could be marginally simplified...
```if (eventData?.Url .... == true)```